tests/test_external_strategies.py

```diff
@@ -0,0 +1,119 @@
+from d3a_interface.constants_limits import GlobalConfig
+from pendulum import now, Duration
+from pytest import mark
+
+
+def ext_strategy_fixture(strategy):
+    config = MagicMock()
+    config.ticks_per_slot = 90
+    GlobalConfig.end_date = GlobalConfig.start_date + Duration(days=1)
+    area = Area(name="forecast_pv", config=config, strategy=strategy,
+                external_connection_available=True)
+    parent = Area(name="parent_area", children=[area])
+    parent.activate()
+    area.activate()
+    strategy.connected = True
+    market = MagicMock()
+    market.time_slot = GlobalConfig.start_date
+    parent.get_future_market_from_id = lambda _: market
+    area.get_future_market_from_id = lambda _: market
+    return strategy
+
+
+class TestPVForecastExternalStrategy:
+
+    @mark.parametrize("strategy", [ext_strategy_fixture(LoadForecastExternalStrategy()),
+                                   ext_strategy_fixture(PVForecastExternalStrategy())])
+    def test_clear_energy_buffers(self, strategy):
+        strategy.energy_forecast_buffer = {now(): 1}
+        strategy.energy_measurement_buffer = {now(): 1}
+        strategy._clear_energy_buffers()
+        assert strategy.energy_forecast_buffer == {}
+        assert strategy.energy_measurement_buffer == {}
+
+    @mark.parametrize("strategy", [ext_strategy_fixture(LoadForecastExternalStrategy()),
+                                   ext_strategy_fixture(PVForecastExternalStrategy())])
+    def test_update_energy_forecast_measurement_is_called_on_market_cycle(self, strategy):
+        strategy.update_energy_forecast = Mock()
+        strategy.update_energy_measurement = Mock()
+        strategy._clear_energy_buffers = Mock()
+        strategy.event_market_cycle()
+        strategy.update_energy_forecast.assert_called_once()
+        strategy.update_energy_measurement.assert_called_once()
+        strategy._clear_energy_buffers.assert_called_once()
+
+    def test_update_forecast_calls_set_available_energy(self, strategy=ext_strategy_fixture(
+            PVForecastExternalStrategy())):
+        time = strategy.area.next_market.time_slot
+        energy = 1
+        strategy.energy_forecast_buffer = {time: energy}
+        strategy.state.set_available_energy = Mock()
+        strategy.update_energy_forecast()
+        strategy.state.set_available_energy.assert_called_once_with(energy,
+                                                                    time, overwrite=True)
+        # do not call set_available_energy for time_slots in the past
+        time = strategy.area.next_market.time_slot.subtract(minutes=15)
+        strategy.energy_forecast_buffer = {time: energy}
+        strategy.state.set_available_energy.reset_mock()
+        strategy.update_energy_forecast()
+        strategy.state.set_available_energy.assert_not_called()
+
+    def test_update_forecast_calls_set_desired_energy_for(self, strategy=ext_strategy_fixture(
```

Comment:
```suggestion
    def test_update_energy_forecast_calls_set_desired_energy(self, strategy=ext_strategy_fixture(
```
I'm not sure what the `_for` stood for, maybe it was a leftover?